File: apps/user/src/dao/NotificationDao.ts
'use strict'

import {SQS} from "aws-sdk";

const AWS = require("aws-sdk")

class NotificationDao {
    private sqs: SQS
    private readonly notificationQueueUrl: string

    constructor() {
        this.sqs = new AWS.SQS()
        // @ts-ignore
        this.notificationQueueUrl = process.env.NOTIFICATION_QUEUE_URL
    }

    async sendNotification(userId: string, title: string, body: string) {
        const data = {
            userId,
            title,
            body
        }

        await this.sqs.sendMessage({
            QueueUrl: this.notificationQueueUrl,
            MessageBody: JSON.stringify(data)
        }).promise()
    }

}

export {NotificationDao}

File: apps/user/src/dao/notification_dao.py
import json
import os

import boto3
from types_boto3_sqs import SQSClient


class NotificationDao:
    def __init__(self):
        self.sqs: SQSClient = boto3.client("sqs")
        self.notification_queue = os.environ["NOTIFICATION_QUEUE_URL"]

    def send_notification(self, user_id: str, title: str, body: str):
        self.sqs.send_message(
            QueueUrl=self.notification_queue,
            MessageBody=json.dumps({
                "userId": user_id,
                "title": title,
                "body": body
            })
        )
